### PR TITLE
Use code point substring algorithm. (#68)

### PR DESCRIPTION
**Pull request by wanderview**
***
<a href="https://pr-preview.s3.amazonaws.com/WICG/urlpattern/pull/97.html" title="Last updated on Aug 16, 2021, 7:29 PM UTC (50dbe61)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/urlpattern/97/224b453...50dbe61.html" title="Last updated on Aug 16, 2021, 7:29 PM UTC (50dbe61)">Diff</a>

### CODE REVIEW

**Comment by wanderview** [comment]
@domenic Can you PTAL at this to make sure I'm using the new code point substring correctly?  Thanks.

File: spec.bs [hunk 1]
```diff
@@ -771,7 +771,7 @@ To run <dfn>make a component string</dfn> given a [=constructor string parser=]
   1. Let |component start token| be the result of running [=get a safe token=] given |parser| and |parser|'s [=constructor string parser/component start=].
   1. Let |component start input index| be |component start token|'s [=token/index=].
   1. Let |new length| be |token|'s [=token/index=] &minus; |component start input index|.
-  1. Return the substring within |parser|'s [=constructor string parser/input=] starting at |component start input index| with |new length|.
+  1. Return the [=code point substring=] from index |component start input index| with length |new length| within |parser|'s [=constructor string parser/input=].
```
Comment (domenic) [inline spec.bs:774]: I think you could collapse this with the previous line by using `[=code point substring by indices|code point substring=] from indices |component start input index| to |token|'s [=token/index=] inclusive within |parser|'s [=constructor string parser/input=]`.

**Review by domenic — APPROVED** [review]
LGTM with minor suggestion

**Comment by wanderview** [comment]
Updated to uses indices.

**Pull request merged**